shm/src/ship.cpp

```diff
@@ -0,0 +1,58 @@
+#include "ship.hpp"
+
+#include <numeric>
+
+Ship::Ship()
+    : Ship(0, 0, 0, "", 0)
+{}
+Ship::Ship(int maxCrew, int speed, size_t id)
+    : Ship(0, maxCrew, speed, "", id)
+{}
+Ship::Ship(int capacity, int maxCrew, int speed,
+        const std::string& name, size_t id)
+    : capacity_(capacity)
+    , maxCrew_(maxCrew) 
+    , crew_(0)
+    , speed_(speed)
+    , name_(name)
+    , id_(id)
+{}
+
+Ship& Ship::operator-=(const size_t crew) {
+    if (crew <= crew_) {
+        crew_ -= crew;
+        return *this;
+    }
+    return *this;
+}

```

Comment:
What about situation when size of crew would become negative. And there should be newline between these two operator definitions.

Reply:
Not sure what you mean.
The size of the crew won't be less then 0 because of the condition applied. But maybe you meant the negative input (e.g. ship -= -10). In such case if an unsigned number is accepted as an argument (size_t here), you won't ever be able to tell unambiguosly if a number is negative (and throw an exception). The client of the class should pass correct values. Alternatively the class can be redesigned to use signed types. Any other ideas are welcome!

Reply:
If an integer type used instead of size_t as an argument the following error/warning appears:
error: comparison of integer expressions of different signedness: 'const int' and 'size_t'
(compiled with -Werror flag)